chapters/05-raster-vector.qmd

```diff
@@ -1,7 +1,797 @@
+ax.ygridvisible = false
+fig # TODO: make GeoMakie better on small extents
+```
+
+<!-- jn: what is the state of plotting categorical rasters? can it read the color palette from a file? -->
+<!-- md: admittedly I've never used this functionality in either R or Python... If you have a sample data file I'll be happy to experiment with it. -->
+<!-- as: We haven't implemented this in **Rasters.jl** yet but it would not be hard potentially... -->
+
+
+## Rasterization {#sec-rasterization}
+
+Rasterization is the conversion of vector objects into their representation in raster objects.
+Usually, the output raster is used for quantitative analysis (e.g., analysis of terrain) or modeling.
+As we saw in @sec-spatial-class, the raster data model has some characteristics that make it conducive to certain methods.
+Furthermore, the process of rasterization can help simplify datasets because the resulting values all have the same spatial resolution: rasterization can be seen as a special type of geographic data aggregation.
+
+**Rasters.jl** provides the `Rasters.rasterize` function for rasterizing vector data.  To make this happen, we need to have some definition for a "template" grid, i.e., the "template" raster defining the extent, resolution and CRS of the output.  We can also pass a pre-existing raster, in which case Rasters uses the existing grid definition.
+
+
+As for the vector geometries and their associated values, the `Rasters.rasterize` function can take input in multiple ways:
+-    `; data::FeatureCollection, fill::Symbol` means that the values of the column passed to `fill` will be used to fill the raster.
+-    `; data::Vector{Geometry}, fill::Vector` means that the values passed to `fill` will be associated with each geometry in `data` and used to fill the raster.
+-    `; data::Any, fill::Function` calls `fill` with the current value
+
+We will use the `; data::FeatureCollection, fill::Symbol` method to rasterize the `zion` polygon we have been using in this chapter.
+
+
+Furthermore, we define how to deal with multiple values burned into the same pixel, in the first argument called `reducer`.  By default, this is `last`, meaning that the last polygon to be rasterized takes precedence.  However, we can pass any function that takes in an iterable and returns a single value, like `mean`, `minimum`, `maximum`, `std`, `median`, `mode`, `sum`, `prod`, etc.  Many of these may not make sense to use but they are useful to know of.
+
+
+Finally, we can set the `fill` value, which is the value that "unaffected" pixels get, with `fill=0` being the default.
+
+How the `Rasters.rasterize` function works with all of these various parameters will be made clear in the next examples.
+
+The geographic resolution of the "template" raster has a major impact on the results: if it is too low (cell size is too large), the result may miss the full geographic variability of the vector data; if it is too high, computational times may be excessive.
```

Comment:
"The geographic resolution" to "the spatial resolution"?

Reply:
```suggestion
The spatial resolution (area of each "pixel") of the "template" raster has a major impact on the results: if it is too low (cell size is too large), the result may miss the full geographic variability of the vector data; if it is too high, computational times may be excessive.